test/__init__.py

```diff
@@ -231,6 +232,17 @@ def _connect(self, host, port, **kwargs):
 
     def _init_client(self):
         self.client = self._connect(host, port)
+
+        if self.client is not None:
+            build_info = self.client.admin.command('buildInfo')
+            if 'dataLake' in build_info:
+                self.data_lake = True
+                self.auth_enabled = True
+                self.client = self._connect(
+                    host, port, username=db_user, password=db_pwd)
+                self.connected = True
+                return
```

Comment:
This seems like something that should run _after_ `_init_client()` returns. Is there a reason it is placed here?

Reply:
`_init_client()` relies on DB commands like `getCmdLineOpts` that are not supported by the `mongohoused` binary.

Reply:
Can you add a comment about that? And how does this work if mongohoused is configured with tls? Do we not test tls?

Reply:
We don't test TLS. And done.